stjude/proteinpaint
# attempt to recache if a caching error is detected besides a stale ver…

…sion; catch all awaited functions that are not in a try-catch block

Requires related branches from sjpp and gdc: `git unite git unite gdc-await-catch`

## Description

Since GDC relies on network requests as part of dataset initialization, it is more prone to startup errors compared to local file-based datasets. This PR branch tries to recover from all initial GDC caching errors, by setting a `ds.__gdc.cacheError` string that the stale cache detector can use in addition to the GDC API version query. 

TODO later: use a different test server or container process for GDC and any network-based datasets in the future?

Test steps:
1. set `serverconfig.gdcCacheCheckWait` to 5-10 seconds
2. turn off dev machine wifi
3. restart `npm run start`
4. when the server log starts to listen to port 3000, and there are GDC-related caching messages, turn the wifi back on, the gdc caching should try recover
5. before the caching completes (can be fast if `extApiCache` is used), turn the wifi off again. This simulates an error that occurs after the GDC API returns a healthy status and before the caching is complete. The caching should hang again.
6. Turn your wifi back on again. The caching should restart with a message like `detected caching error: cacheMappingOnNewRelease() call to fetchIdsFromGdcApi(), getCasesWithGeneExpression(), or getAnalysisTsv2loom4scrna()`

`cached gdc data version is up-to-date, but there was a caching error, will restart cache`

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
